### DISCUSSION TITLE
How is Net Worth calculated?

### DISCUSSION THREAD

**Discussion by arn4** [firefly-iii/firefly-iii#6120] [Firefly III]
I'm asking this question because I've noticed that the "Net worth" displayed in the main page is different from the sum of assets account I have. 
I'm sure that all accounts have "Include in net worth" checked. 
I'm using two currencies, I do not know if it matters

**Comment by JC5** [comment]
It's a basic sum, but i can check out the code.

**Comment by arn4** [comment]
More information: I’ve noticed that the plot of Net Worth even goes negative in some periods (this does not make any sense, even if I’m missing something in the “include in Net Worth” flag).
I can privately send you detailed informations on the plots ands the accounts, if you need them to investigate.

**Reply by JC5** [reply]
Which net worth are you talking about exactly?

**Reply by arn4** [reply]
In the homepage on the top right there is a box displaying the Net Worth. If you press on it appears a report, that it seems different from the default financial report, where you can find a plot of the Net Worth along the period. In that plot sometimes my Net Worth in CHF goes below zero, that does not make sense because I've never been negative in my bank accounts

**Comment by arn4** [comment]
More information: I’ve noticed that the plot of Net Worth even goes negative in some periods (this does not make any sense, even if I’m missing something in the “include in Net Worth” flag).
I can privately send you detailed informations on the plots ands the accounts, if you need them to investigate

**Reply by JC5** [reply]
The basic calculation can be found [here](https://github.com/firefly-iii/firefly-iii/blob/main/app/Helpers/Report/NetWorth.php#L68) and is pretty simple.

- Collect the accounts [here](https://github.com/firefly-iii/firefly-iii/blob/main/app/Http/Controllers/Json/BoxController.php#L217)
- Grab the latest balance for each account. Sum them up. Add or remove virtual balance. 
- Done!

**Comment by tmllull** [comment]
EDIT: I think I found the "problem" with the graph: it not shows the last days. If I generate en entire year report, the graph shows the next months (with no transactions), and the end value is correct (shows a flat line for the next months). But if I generate a report from the first day until today, not shows the last transactions. If today is November 29, the graph only shows until November 26, and "lose" these last transactions.

Report dates
![imagen](https://user-images.githubusercontent.com/5560652/204502853-65698a37-6694-4ab4-90d6-6500135fd5ca.png)

Graph last date
![imagen](https://user-images.githubusercontent.com/5560652/204502895-58e7fe86-b51a-400c-916c-cc323323bf03.png)

I deleted the original message because not apply after the edit.

**Reply by JC5** [reply]
What is the URL of the report?

**Reply by tmllull** [reply]
From report section, selecting "All-time, All accounts" (wrong graph): MYURL/reports/default/19,22,21/20220101/currentMonthEnd
![imagen](https://user-images.githubusercontent.com/5560652/204506909-7f00d6e4-92c7-472b-8fc7-ca0cfcc894f6.png)

From home, clicking on Net Worth label: MYURL/reports/default/allAssetAccounts/currentYearStart/currentYearEnd
![imagen](https://user-images.githubusercontent.com/5560652/204506942-c5dd2e51-7476-41ed-b768-ecec24210b76.png)

**Reply by tmllull** [reply]
Seems that the graph shows the values based on weeks (every 7 days). As this month ends in the middle of the week, shows the last "week report" (Nov 26), because on the entire year report, the next bullet on the graph is December 3 (7 days later). So, the transactions between Nov 26 to 30 not appears in the graph (but yes in the numeric values).
